src/lib/core/components/visualizations/implementations/HistogramVisualization.tsx

```diff
@@ -366,6 +379,54 @@ function HistogramViz(props: VisualizationProps) {
         }
       : undefined;
 
+  // custom legend items for checkbox
+  const legendItems: LegendItemsProps[] = useMemo(() => {
+    const legendData = !isFaceted(data.value)
+      ? data.value?.series
+      : data.value?.facets[0].data.series;
+
+    return legendData != null
+      ? legendData
+          .map((dataItem: HistogramDataSeries, index: number) => {
+            return {
+              label: dataItem.name,
+              // histogram plot does not have mode, so set to square for now
+              marker: 'square',
+              markerColor:
+                dataItem.name === 'No data'
+                  ? '#E8E8E8'
+                  : ColorPaletteDefault[index],
+              // deep comparison is required for faceted plot
+              hasData: !isFaceted(data.value) // no faceted plot
+                ? dataItem.bins != null && dataItem.bins.length > 0
+                  ? true
+                  : false
+                : data.value?.facets // faceted plot: here data.value is full data
+                    .map(
+                      (el: { label: string; data: HistogramData }) =>
+                        el.data.series[index].bins != null &&
+                        el.data.series[index].bins.length > 0
+                    )
+                    .includes(true)
+                ? true
+                : false,
+              group: 1,
+              rank: 1,
+            };
+            // histogram viz uses stack option so reverse the legend order!
+          })
+          .reverse()
+      : [];
+  }, [data]);
+
+  // use this to set all checked
+  useEffect(() => {
+    if (data != null) {
+      // use this to set all checked
+      onCheckedLegendItemsChange(legendItems.map((item) => item.label));
+    }
+  }, [data, legendItems]);
```

Comment:
@dmfalke thank you for your suggestion. Actually I am putting that work in the future (possibly next release as I have several things to do) as it would be quite tricky one to make appropriate conditions that cover up all use cases. To accomplish it, it should be like: 

```
useEffect(() => {
...
(if condition A) - generate checkedLegendItems
onCheckedLegendItemsChange(legendItems.map((item) => item.label));
(else/else if condition B) - this will keep previous checkedLegendItems for a use case of leaving the viz and coming back to the one
onCheckedLegendItemsChange(legendItems.map(vizConfig.checkedLegendItems);
...
```

Although there are a couple of more use cases that your suggestion is not working, One simple use case is when a user **changes** variable (X, Y, Main, or Overlay) from existing one - since vizConfig.checkedLegendItems was previously defined in this case, checkedLegendItems does not change because onCheckedLegendItemsChange is not called.

Reply:
That makes sense. I'm fine with putting this off, then.